### ISSUE TITLE
Even though some posts have no "Featured Image" set, `if ( has_post_thumbnail() )` still triggers

### ISSUE THREAD

**Issue by andrebu** [iamandrebulatov/Theme-Redesign-and-SEO-Overhaul#35]
Even though some posts have no "Featured Image" set, `if ( has_post_thumbnail() )` still triggers, causing an empty box where an image or its fill-in fallback should be:

<img width="1280" alt="screen shot 2016-02-23 at 12 08 33 am" src="https://cloud.githubusercontent.com/assets/3777877/13242349/da9b7d2a-d9c1-11e5-8d0d-30a4e77a19f9.png">

**Comment by andrebu** [comment]
The post use to have a Featured or thumbnail image, but something got messed up or image was deleted, and the image URL would redirect to the site's fall-back `http://images.earmilk.com/delivery.png'`.

Checking for `has_post_thumbnail()` returns a `1` or `true`, and yet no image exists.  So instead of outputting an `<img>` tag with fallback image URL, it will simply output a `the_post_thumbnail()` as if all is normal, except that said tag will be empty.

**Comment by andrebu** [comment]
The Fix

functions.php:

``` php
// ---------------------------------------------------------------------------------------------------- 
// Album Review titles -- append "Album Review:" to titles - Andre
// ---------------------------------------------------------------------------------------------------- 
function fallback_thumbnail_image( $tag_or_src ) {
    $thumbnail_fallback_tag = '<img src="http://images.earmilk.com/delivery.png" alt="Really hot thumbnail for this article post!" />';
    $thumbnail_fallback_src = 'http://images.earmilk.com/delivery.png';
    $response = '';
    if ( $tag_or_src == 'src' ) {
        $response = $thumbnail_fallback_src;
    } elseif ( $tag_or_src == 'tag' ) {
        $response = $thumbnail_fallback_tag;
    } else {
        $response = $thumbnail_fallback_tag;
    }
    return $response;
}
```

and in index.php:

``` html+php
<?php ... ?>
...
<div itemprop="image" itemscope itemtype="https://schema.org/ImageObject">
    <?php
        $post_thumbnail_id = get_post_thumbnail_id();
        $post_thumbnail_url = wp_get_attachment_url( $post_thumbnail_id );
    ?>
    <?php if ( has_post_thumbnail() ) { ?> 
        <?php if ( get_the_post_thumbnail() ) { ?> 
            <?php echo the_post_thumbnail('thumbnail-blog'); ?>
            <meta itemprop="url" content="<?php echo $post_thumbnail_url; ?>" />
        <?php } else { ?>
            <?php echo fallback_thumbnail_image(); ?>
            <meta itemprop="url" content="<?php echo fallback_thumbnail_image('src'); ?>" />
        <?php } // Post with messed up Thumbnail ?>
    <?php } else { ?>
        <!-- post has no thumbnail -->
        <?php echo fallback_thumbnail_image(); ?>
        <meta itemprop="url" content="<?php echo fallback_thumbnail_image('src'); ?>" />
    <?php } // Post Thumbnail ?> 
    <meta itemprop="width" content="283" />
    <meta itemprop="height" content="133" />
</div>
...
<?php ... ?>
```

**Comment by montreyw** [comment]
THANK YOU THANK YOU. Tried to find solutions for years.

Montrey Whittaker
Co-Founder / President

_808.673.0404_
_earmilk.com http://earmilk.com/_
_twitter.com/earmilk http://twitter.com/earmilk_

_facebook.com/earmilk http://facebook.com/earmilk__instagram.com/earmilk_
http://instagram.com/earmilk

**Comment by montreyw** [comment]
[23-Feb-2016 14:38:31 UTC] PHP Warning:  Missing argument 1 for
fallback_thumbnail_image(), called in
/home/www-data/earmilk/public/wp-content/themes/mag-wp/index.php on line
189 and defined in
/home/www-data/earmilk/public/wp-content/themes/mag-wp/functions.php on
line 534
[23-Feb-2016 14:38:31 UTC] PHP Warning:  Missing argument 1 for
fallback_thumbnail_image(), called in
/home/www-data/earmilk/public/wp-content/themes/mag-wp/index.php on line
184 and defined in
/home/www-data/earmilk/public/wp-content/themes/mag-wp/functions.php on
line 534
[23-Feb-2016 14:38:31 UTC] PHP Warning:  Missing argument 1 for
fallback_thumbnail_image(), called in
/home/www-data/earmilk/public/wp-content/themes/mag-wp/index.php on line
189 and defined in
/home/www-data/earmilk/public/wp-content/themes/mag-wp/functions.php on
line 534
[23-Feb-2016 14:38:31 UTC] PHP Warning:  Missing argument 1 for
fallback_thumbnail_image(), called in
/home/www-data/earmilk/public/wp-content/themes/mag-wp/index.php on line
189 and defined in
/home/www-data/earmilk/public/wp-content/themes/mag-wp/functions.php on
line 534
[23-Feb-2016 14:38:31 UTC] PHP Warning:  Missing argument 1 for
fallback_thumbnail_image(), called in
/home/www-data/earmilk/public/wp-content/themes/mag-wp/index.php on line
184 and defined in
/home/www-data/earmilk/public/wp-content/themes/mag-wp/functions.php on
line 534

Montrey Whittaker
Co-Founder / President

_808.673.0404_
_earmilk.com http://earmilk.com/_
_twitter.com/earmilk http://twitter.com/earmilk_

_facebook.com/earmilk http://facebook.com/earmilk__instagram.com/earmilk_
http://instagram.com/earmilk

**Comment by andrebu** [comment]
Fixed #35-(b) index.php [reference here](https://github.com/iamandrebulatov/Theme-Redesign-and-SEO-Overhaul/commit/76032886fe00480a61eafe08e7c38b6c0ebbb26b): 

``` php
        // like this 
    <?php echo fallback_thumbnail_image('tag'); ?>
        // instead of this
    <?php echo fallback_thumbnail_image(); ?>
```

....and made function work [without parameter by default here](https://github.com/iamandrebulatov/Theme-Redesign-and-SEO-Overhaul/commit/c78d7cd19d05c035d68680e65c973a1f2bcff5fc), like so:

``` php
// ---------------------------------------------------------------------------------------------------- 
// Fallback thumbnail image tag and src function  - Andre
// ---------------------------------------------------------------------------------------------------- 
function fallback_thumbnail_image( $tag_or_src = 'tag' ) {
    $thumbnail_fallback_tag = '<img src="http://images.earmilk.com/delivery.png" alt="Really hot thumbnail for this article post!" />';
    $thumbnail_fallback_src = 'http://images.earmilk.com/delivery.png';
    $response = '';
    if ( $tag_or_src == 'src' ) {
        $response = $thumbnail_fallback_src;
    } elseif ( $tag_or_src == 'tag' ) {
        $response = $thumbnail_fallback_tag;
    } else {
        $response = $thumbnail_fallback_tag;
    }
    return $response;
}
```
